### PR TITLE
Proofpoint TAP v2 - short interval bug

### PR DESCRIPTION
**Pull request by ChanochShayner**
…d interval*.

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29795

## Description
Fixed an issue where fetch-incident failed on too short request interval.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ChanochShayner** [comment]
@jochman 
Should the integration be sent for testing to the customer to make sure the problem is resolved?

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def get_fetch_times(last_fetch):
     elif isinstance(last_fetch, datetime):
         times.append(last_fetch.strftime(time_format))
     while now - last_fetch > timedelta(minutes=59):
-        last_fetch += timedelta(minutes=59)
+        last_fetch += timedelta(minutes=58)
```
Comment (ronykoz) [inline Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:55]: it failed on too short and you are making it shorter?

Comment (ChanochShayner) [reply Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:55]: It fails at a 30-second interval, 
and the change I made makes sure the shortest interval that can be is a minute.

**Comment by ChanochShayner** [comment]
The test failed to check the number of intervals that will enter the list -
As in issue - a 10-second interval enters the list.
https://app.circleci.com/jobs/github/demisto/content/160332
image

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py (hunk 1 above)
Comment (jochman) [reply Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:55]: you shouldn't change it as checks for the longest interval

**Comment by jochman** [comment]
> @jochman
> Should the integration be sent for testing to the customer to make sure the problem is resolved?

You can test is with unit-tests only as the bug is well defined

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py (hunk 1 above)
Comment (jochman) [reply Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:55]: or: add an informative comment on why it's 58 minutes.

**Review by jochman — CHANGES_REQUESTED** [review]
Add a comment of how/why the interval changed to 58 minutes.

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def get_fetch_times(last_fetch):
         last_fetch = datetime.strptime(last_fetch, time_format)
     elif isinstance(last_fetch, datetime):
         times.append(last_fetch.strftime(time_format))
-    while now - last_fetch > timedelta(minutes=59):
+    while now - last_fetch > timedelta(minutes=59) and now - last_fetch > timedelta(minutes=60):
```
Comment (guykeller) [inline Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:54]: why add another condition? The first condition is included in the second.
Also, why does this solve the bug?

**Review by guykeller — CHANGES_REQUESTED** [review]

**Comment by ChanochShayner** [comment]
According to the issue the problem is not solved, it seems that the problem is not solved (even though the "get_fetch_times" function certainly does not return times if the differences are less than a minute).
@jochman do you have an idea what else can cause the problem?

**Comment by ChanochShayner** [comment]
@jochman The customer reports that the error no longer occurs, but it happens to him from time to time (once or twice a day) the following error - 
image
Is this a known error in this integration?

**Comment by jochman** [comment]
@ChanochShayner what with this pr?

**Pull request closed without merging**